lib/rex/post/postgresql/ui/console/interactive_channel.rb

```diff
@@ -0,0 +1,160 @@
+# -*- coding: binary -*-
+module Rex
+module Post
+module PostgreSQL
+module Ui
+
+###
+#
+# Mixin that is meant to extend the base channel class from meterpreter in a
+# manner that adds interactive capabilities.
+#
+###
+module Console::InteractiveChannel
+
+  include Rex::Ui::Interactive
+
+  #
+  # Interacts with self.
+  #
+  def _interact
+    begin
+      while self.interacting
+        raw_sql_string = _get_user_input
+        # We need to check that the user is still interacting, i.e. if ctrl+z is triggered when requesting user input
+        break unless (self.interacting && raw_sql_string)
+
+        formatted_query = client_dispatcher.process_query(query: raw_sql_string)
+        print_status "Executing query: #{formatted_query}"
+        client_dispatcher.cmd_query(formatted_query)
+      end
+    rescue ::StandardError => e
+      elog('Error when interact with SQL REPL', e)
+      raise e
+    end
```

Comment:
I think we're probably good to remove this; it's probably logged up a level already - worth checking